### PR TITLE
feat:Provide org service address (CMC-1659)

### PR DESCRIPTION
**Pull request by dharmendrak**
### JIRA link (if applicable) ###

CMC-1659

### Change description ###

Provide org service address for claimant and defendant solicitors

Screenshot 2021-06-17 at 09 11 30
Screenshot 2021-06-17 at 09 13 13

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes

[ ] No
```

### CODE REVIEW

File: ccd-definition/CaseField.json [hunk 1]
```diff
@@ -542,6 +542,13 @@
     "FieldType": "OrganisationPolicy",
     "SecurityClassification": "Public"
   },
+  {
+    "CaseTypeID": "CIVIL",
+    "ID": "applicantSolicitor1ServiceAddress",
```
Comment (pd-dudkiewicz) [inline ccd-definition/CaseField.json:547]: we need new fields in CaseData model in civil-service repo before merging this PR

File: e2e/fixtures/events/createClaim.js [hunk 2]
```diff
@@ -92,6 +98,12 @@ const createClaimData = (legalRepresentation, useValidPba) => {
         },
       },
     },
+    DefendantSolicitorServiceAddress: {
```
Comment (pd-dudkiewicz) [inline e2e/fixtures/events/createClaim.js:101]: We should also include `SecondDefendantSolicitorServiceAddress` page in api tests. See how it's done for `AddAnotherClaimant`.

File: e2e/pages/createClaim/claimantSolicitorServiceAddress.page.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+const address = require('./../../fixtures/address.js');
+const postcodeLookup = require('./../../fragments/addressPostcodeLookup');
+
+const {I} = inject();
+
+module.exports = {
+
+  fields: {
+    applicantSolicitor1ServiceAddress_hasServiceAddress: {
```
Comment (pd-dudkiewicz) [inline e2e/pages/createClaim/claimantSolicitorServiceAddress.page.js:9]: How about `applicantSolicitor1HasServiceAddress`? We shouldn't mix camel case with underscores in property names.

File: e2e/pages/createClaim/claimantSolicitorServiceAddress.page.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  },
+
+  async enterOrganisationServiceAddress() {
+    I.waitForElement(this.fields.applicantSolicitor1ServiceAddress_hasServiceAddress.id);
+    await I.runAccessibilityTest();
+    await within(this.fields.applicantSolicitor1ServiceAddress_hasServiceAddress.id, () => {
+      I.click(this.fields.applicantSolicitor1ServiceAddress_hasServiceAddress.options.yes);
+    });
+
+    postcodeLookup.enterAddressManually(address);
```
Comment (pd-dudkiewicz) [inline e2e/pages/createClaim/claimantSolicitorServiceAddress.page.js:25]: this should be wrapped in `await within(...)`, see `litigationFriend.js` as an example

File: ccd-definition/CaseEventToComplexTypes/CreateClaim-nonprod.json [hunk 5]
```diff
@@ -170,5 +170,69 @@
     "ListElementCode": "OrgPolicyReference",
     "FieldDisplayOrder": 4,
     "DisplayContext": "OPTIONAL"
+  },
+  {
+    "ID": "Respondent2ServiceAddress",
+    "CaseEventID": "CREATE_CLAIM",
+    "CaseFieldID": "respondentSolicitor2ServiceAddress",
+    "ListElementCode": "hasServiceAddress",
```
Comment (swalker125) [inline ccd-definition/CaseEventToComplexTypes/CreateClaim-nonprod.json:178]: I would maybe just call this field required. Then it's respondentSolicitor2ServiceAddress.required

File: e2e/pages/createClaim/defendantSolicitorServiceAddress.page.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
+const { I } = inject();
+
+const address = require('./../../fixtures/address.js');
+const postcodeLookup = require('./../../fragments/addressPostcodeLookup');
+
+module.exports = {
+
+  fields: {
+    respondentSolicitor1ServiceAddress_hasServiceAddress: {
```
Comment (pd-dudkiewicz) [inline e2e/pages/createClaim/defendantSolicitorServiceAddress.page.js:9]: as mentioned in `claimantSolicitorServiceAddress.page.js `

File: e2e/pages/createClaim/defendantSolicitorServiceAddress.page.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  },
+
+  async enterOrganisationServiceAddress() {
+    I.waitForElement(this.fields.respondentSolicitor1ServiceAddress_hasServiceAddress.id);
+    await I.runAccessibilityTest();
+    await within(this.fields.respondentSolicitor1ServiceAddress_hasServiceAddress.id, () => {
+      I.click(this.fields.respondentSolicitor1ServiceAddress_hasServiceAddress.options.yes);
+    });
+
+    postcodeLookup.enterAddressManually(address);
```
Comment (pd-dudkiewicz) [inline e2e/pages/createClaim/defendantSolicitorServiceAddress.page.js:25]: as mentioned in `claimantSolicitorServiceAddress.page.js `

**Review by swalker125 — COMMENTED** [review]
One comment about field naming, otherwise LGTM

File: e2e/steps_file.js [hunk 8]
```diff
@@ -146,6 +149,7 @@ module.exports = function () {
         ... conditionalSteps(!litigantInPerson, [
           () => respondentRepresentedPage.enterRespondentRepresented('respondent1', 'yes'),
           () => defendantSolicitorOrganisation.enterOrganisationDetails('respondent1'),
+          () => defendantSolicitorServiceAddress.enterOrganisationServiceAddress(),
```
Comment (pd-dudkiewicz) [inline e2e/steps_file.js:152]: `SecondDefendantSolicitorServiceAddress` page needs to be covered on e2e tests as well

File: ccd-definition/CaseEventToComplexTypes/CreateClaim-nonprod.json (hunk 5 above)
Comment (dharmendrak) [reply ccd-definition/CaseEventToComplexTypes/CreateClaim-nonprod.json]: actioned

File: ccd-definition/CaseField.json (hunk 1 above)
Comment (dharmendrak) [reply ccd-definition/CaseField.json:547]: got civil service  PR ready

File: e2e/steps_file.js (hunk 8 above)
Comment (dharmendrak) [reply e2e/steps_file.js:153]: done

**Review by douglasrice — APPROVED** [review]

**Review by swalker125 — APPROVED** [review]

**Pull request merged**